File: BA3/enterprise-applications/project/API/src/main/java/art/ameliah/ehb/anki/api/models/deck/Card.java
package art.ameliah.ehb.anki.api.models.deck;

import io.ebean.Model;
import io.ebean.annotation.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

import jakarta.persistence.Entity;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Card extends Model {

    @Id
    Long id;

    @NotNull
    @Enumerated
    CardType type;

    @NotNull
    @Enumerated
    Difficulty difficulty;

    @NotNull
    String question;

    @NotNull
    String hint;

    @NotNull
    String information;

    @ManyToOne
    Deck deck;


    public enum CardType {
        STANDARD,
        MULTI
    }

    public enum Difficulty {
        EASY,
        MEDIUM,
        HARD,
        IMPOSSIBLE
    }

}

File: BA3/enterprise-applications/project/UI/Web/src/app/_models/card.ts

export interface Card {
  id: number;
  deckId: number;
  type: CardType;
  difficulty: Difficulty;
  question: string;
  hint?: string;
  information: string;
  answers: {id: number, answer: string, correct: boolean}[];
}

export enum CardType {
  STANDARD= "STANDARD",
  MULTI = "MULTI",
}

export const cardTypes = Object.keys(CardType);

export enum Difficulty {
  EASY = "EASY",
  MEDIUM = "MEDIUM",
  HARD = "HARD",
  IMPOSSIBLE = "IMPOSSIBLE",
}

export const difficulties = Object.keys(Difficulty)

export interface CreateCard {
  type: CardType;
  difficulty: Difficulty;
  question: string;
  hint: string;
  information: string;
  deck: number;
}
